src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -405,6 +416,16 @@ public void SetHttpServerErrorStatusCodes(IEnumerable<int> statusCodes)
             HttpServerErrorStatusCodes = ParseHttpCodesToArray(string.Join(",", statusCodes));
         }
 
+        /// <summary>
+        /// Sets a mapping to use as the service name within a <see cref="Span"/>
+        /// </summary>
+        /// <param name="originalName">The original name of the service, for example <code>sql-server</code> or <code>graphql</code> </param>
+        /// <param name="newName">The new name to use for the service</param>
+        public void SetServiceNameMapping(string originalName, string newName)
```

Comment:
Personally I would have opted for a `IEnumerable<KeyValuePair<string, string>>`, which has the benefit of overriding the values that were loaded from the environment variable. But I don't have strong feelings about it, I can find pros for both approaches.

Reply:
I don't have strong feelings about it either, happy to change to that 🙂

> which has the benefit of overriding the values that were loaded from the environment variable.

Ah, yes, currently there's no way to _remove_ values added using the environment variable, you can only add or replace them. I'll switch to your suggestion instead. 

Only downside would be there's no way for public users to _add_ to the existing list of service mappings (without removing existing ones), as we don't expose that list currently

Reply:
> Only downside would be there's no way for public users to _add_ to the existing list of service mappings (without removing existing ones), as we don't expose that list currently

Yeah but I doubt that's a use-case. If I'm setting a configuration value from the code, I do not expect it to be merged with an environment variable 🤔 It could be just me though.

Reply:
Meh, I don't know, I could see setting host-wide variables, but wanting an override for a specific application🤷‍♂️ I'll go with your approach though, as it's probably more likely 👍

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/1142/commits/e2c02a24bbd27f5d9b9ae888a76cbf9f78dafb4c